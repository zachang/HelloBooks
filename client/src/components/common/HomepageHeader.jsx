import React from 'react';
import { Link, IndexLink } from 'react-router';

export default class HomepageHeader extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            regLoginToggle: true
        };
        this.handleToggle = this.handleToggle.bind(this);
    }

    componentDidMount() {
        $('.button-collapse').sideNav();
    }

    handleToggle(e) {
        const value = (e.currentTarget.getAttribute('data-toggle') === 'login');
        this.props.regLoginToggle(value);
    }

    render() {
        return (
            <div className="navbar-fixed">
                <nav className="mainNav transparent" role="navigation">
                    <div className="nav-wrapper container">
                        <a href="#" onClick={this.handleToggle} data-toggle="login" id="logo-container" className="brand-logo mainLogo brand">
                            <img src="./imgs/hello.png" alt="hellobooks"/>
                        </a>
                        <ul className="right hide-on-small-and-down">
                            <li><Link to="/">Contact us</Link></li>
                            <li><Link to="/">About us</Link></li>
                            <li><Link to="/">Help</Link></li>
                            <li>
                                <a href="#" onClick={this.handleToggle} data-toggle="toggle" className="btn-large waves-effect waves-light orange join-us">
                                    {(
                                      (this.props.btnText === true) ?
                                        'Join Us': 'Login'
                                    )}
                                </a>
                            </li>
                        </ul>

                        <ul id="nav-mobile" className="side-nav hide-on-med-and-up">
                            <li><Link to="/">Contact us</Link></li>
                            <li><Link to="/">Contact us</Link></li>
                            <li><Link to="/">Contact us</Link></li>
                            <li><Link to="/login">Login</Link></li>
                            <li>
                                <Link to="/register" id="join" className="btn-large waves-effect waves-light orange" style={{ margin: '0px', padding: '0px' }}>Join us</Link>
                            </li>
                        </ul>
                        <a href="#" data-activates="nav-mobile" className="button-collapse full top-nav hide-on-med-and-up menu-iconic-link">
                            <i className="material-icons">menu</i>
                        </a>
                    </div>
                </nav>
            </div>
        );
    }
}