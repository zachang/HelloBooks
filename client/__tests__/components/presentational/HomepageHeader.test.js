import React from 'react';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { Router, browserHistory } from 'react-router';
import { shallow } from 'enzyme';

import HomepageHeader from '../../../src/components/common/HomepageHeader.jsx';

const props = {
  btnText: false
};

const shallowWrapper = shallow(
  <Router history={browserHistory}>
    <HomepageHeader {...props}/>
  </Router>
);
const tree = toJson(shallowWrapper);
let wrapper;

describe('<HomepageHeader/>', () => {
  it('renders <HomepageHeader /> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('calls componentDidMount', () => {
    sinon.spy(HomepageHeader.prototype, 'componentDidMount');
    wrapper = shallow(<HomepageHeader {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(HomepageHeader.prototype.componentDidMount.calledOnce).toEqual(true);
  });
});

