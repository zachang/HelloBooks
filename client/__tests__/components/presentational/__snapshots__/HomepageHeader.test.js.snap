// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomepageHeader/> renders <HomepageHeader /> component 1`] = `""`;
